File: SpringBoot-Pi-main/src/main/java/com/example/pokerplanninpi/entity/Reclamation.java
package com.example.pokerplanninpi.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;
import java.time.LocalDate;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Reclamation {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;

    private String name;
    private String descriptionReclamation;
    private LocalDate dateSoumission;
    @Enumerated(EnumType.STRING)
    private CategorieType categorie;

    @JsonIgnore
    @OneToMany(mappedBy = "reclamation", cascade = CascadeType.ALL)
    private List<ReponseReclamation> reponses;

    enum CategorieType {
        Missing_Functionalities,
        Technical_Problems,
        Security,
        Support_and_Assitance,
        Performances
    }




    @ManyToOne
    User user;

    @Enumerated(EnumType.STRING)
    private StatusType status = StatusType.NotProcessed;

    public  enum StatusType {
        NotProcessed,
        Processed
    }

    public String getLogin(){
        return "bensalah.saif@esprit.tn";
    }
}

File: AngluarPI-main/src/app/model/reclamation.ts
import { User } from "./User";

export enum CategorieType {
    Missing_Functionalities,
    Technical_Problems,
    Security,
    Support_and_Assitance,
    Performances
}
export class Reclamation {
    constructor(

        public id?: number,
        public descriptionReclamation?: string,
        public name?: string,
        public dateSoumission?: Date,
        public categorie?: CategorieType,
        public user?: User,
        public status?: string 

    ) {}
    }
  
    